src/components/Tree/Tree.test.tsx

```diff
@@ -0,0 +1,191 @@
+/**
+ * Copyright 2024 Mia srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { TreeDataNode } from 'antd'
+
+import { fireEvent, render, screen, waitFor } from '../../test-utils'
+import { Icon } from '../Icon'
+import { Tree } from '.'
+
+const defaultTreeData: TreeDataNode[] = [
+  {
+    title: 'Fruits',
+    key: 'fruits',
+    children: [
+      {
+        title: 'Favorites',
+        key: 'favorites',
+        children: [
+          {
+            title: 'Apple',
+            key: 'apple',
+          },
+          {
+            title: 'Banana',
+            key: 'banana',
+          },
+          {
+            title: 'Peach',
+            key: 'peach',
+          },
+        ],
+      },
+      {
+        title: 'Others',
+        key: 'others',
+        children: [
+          {
+            title: 'Strawberry',
+            key: 'strawberry',
+          },
+          {
+            title: 'Pineapple',
+            key: 'pineapple',
+          },
+          {
+            title: 'Fig',
+            key: 'fig',
+          },
+        ],
+      },
+    ],
+  },
+  {
+    title: 'Vegetables',
+    key: 'vegetables',
+    children: [
+      {
+        title: 'Spinach',
+        key: 'spinach',
+      },
+      {
+        title: 'Leek',
+        key: 'leek',
+      },
+    ],
+  },
+]
+
+describe('Tree Component', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  test('renders Tree', async() => {
+    const { asFragment } = render(<Tree treeData={defaultTreeData} />)
+
+    expect(asFragment()).toMatchSnapshot()
+
+    expect(screen.getByText('Vegetables')).toBeInTheDocument()
+    const [, vegetablesCaret] = screen.getAllByRole('img', { name: /caret-down/i })
+    fireEvent.click(vegetablesCaret)
```

Comment:
I think you should use the "userEvent" library because you should apply some checks on user clicks. Fireevent is the lower API level

Reply:
Nothing against that, expect the warning showing up during the text execution:

```
When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
```

(btw, the `act` imported from `test-utils` folder results deprecated, the `act` included in `react` package does not solve the warning)

Should I go with `userEvent` anyway?

Reply:
mmm, no. let's leave fireevent